src/main/java/spoon/support/reflect/code/CtStatementImpl.java

```diff
@@ -67,10 +67,14 @@ public static void insertBefore(CtStatement target, CtStatementList statementsTo
 		if (targetParent instanceof CtExecutable) {
 			throw new SpoonException("cannot insert in this context (use insertEnd?)");
 		}
-		if (target.getParent(CtConstructor.class) != null) {
-			if (target instanceof CtInvocation && ((CtInvocation<?>) target).getExecutable().getSimpleName().startsWith("<init>")) {
-				throw new SpoonException("cannot insert a statement before a super or this invocation.");
+		try {
+			if (target.getParent(CtConstructor.class) != null) {
+				if (target instanceof CtInvocation && ((CtInvocation<?>) target).getExecutable().getSimpleName().startsWith("<init>")) {
+					throw new SpoonException("cannot insert a statement before a super or this invocation.");
+				}
 			}
+		} catch (ParentNotInitializedException ignore) {
+			// no constructor in parent
```

Comment:
You have this exception only because you set the parent of the CtClass to null when you use the snippet to create the class.